qutip/core/metrics.py

```diff
@@ -75,9 +75,7 @@ def fidelity(A, B):
     --------
     >>> x = fock_dm(5,3)
     >>> y = coherent_dm(5,1)
-    >>> fidelity(x,y)
-    0.24104350624628332
-
+    >>> np.testing.assert_almost_equal(fidelity(x,y), 0.24104350624628332)
```

Comment:
Having doctest like this for only a few functions does not feel right...  
We should think about it when working on the docs.

Reply:
Yeah, it feels like maybe someone got things mixed up when they wrote this doctest - surely the purpose of doctests are meant to check the existing examples make sense, rather than obfuscating the example?  We'll have to work out how to do approximate numeric comparisons, I suppose.